Export schemas

IvanCoronado opened the pull request:
Added "Save model definitions" in the bar menu.
"Save model definitions" options will be enabled just when you have focused a realm browser window.
Languages supported JS and Swift.

closes (partially) #158

IvanCoronado committed:
feat (MainMenu): Add logic for enable/disable model definitions menu options

IvanCoronado committed:
style (MainMenu): Lint

IvanCoronado committed:
feat (MainMenu): Fix ts error

IvanCoronado committed:
Add 'Save model definitions' logic
*Enable/Disable when a realm browser window gains/loses focus
*Export JS
*Export swift

IvanCoronado committed:
Merge branch 'master' into ic/export-schemas

* master:
  Context menu revisited (#426)
  Browser row caching (#425)
  Prepare version v1.7.3
  Removing the override of default ports (#416)
  feat (Cell): Show dates in ISO format (#403)
  feat (realms): Remove timeout while opening realm (#402)

# Conflicts:
#	src/ui/realm-browser/RealmBrowserContainer.tsx

IvanCoronado committed:
feat (RealmBrowserContainer): Improve electron import

kraenhansen reviewed (changes requested):
There's a lot of TS code in the exporter service that I would like to follow the codestyle of the project.

Also yes - I would like the tests to be passing before merging this in.

IvanCoronado committed:
feat (schema-export): Remove TS errors and refactor structure

IvanCoronado committed:
feat (RealmBrowserContainer): Unify export schema functions

IvanCoronado committed:
refactor (MainMenu): Call new exportSchema function

IvanCoronado committed:
refactor (MainMenu): Use curly-brackets in getMenuItem

IvanCoronado committed:
refactor (MainMenu): Rename enableModelDefinitions to enableExportSchemaOption

IvanCoronado committed:
refactor (schemaExporter): Now is typed as abstract

IvanCoronado committed:
refactor (schemaExporter): Now is typed as abstract

IvanCoronado committed:
refactor (schema-export): Simplify schema-export file structure

IvanCoronado committed:
feat (schema-export): Add coverage for JS and Swift

IvanCoronado committed:
feat (RealmBrowserContainer): Move electron dialog logic to MainMenu

IvanCoronado committed:
feat (WindowsManager): createWindow now can receive callbacks for events

IvanCoronado committed:
Merge branch 'master' into ic/export-schemas

* master:
  Prepare version v1.7.4
  Fixing the window titles of ROS administration and Realm Browser (#432)
  Row render revisited (#433)
  Making the server URL selectable (#430)

nirinchev committed:
Update c# expected output

IvanCoronado committed:
feat (schema-export): Use models/sample instead of demo-model object (+1 squashed commit)
Squashed commits:
[31bcabb] feat (schema-export): Use models/sample instead of demo-model object

kraenhansen reviewed (changes requested):
I think it's a little weird that the services/scheme-export has a "src" folder inside it, would it be difficult to just move the code one level up?

IvanCoronado committed:
feat (schema-export): Remove src level in the files hierarchy

IvanCoronado committed:
feat (WindowManager): Rename callbackEvent to eventListenerCallbacks and add IEventListenerCallbacks

IvanCoronado committed:
refactor (RealmBrowserContainer): Move show dialog logic to RealmBrowserContainer

nhachicha committed:
add Java schema exporter

nhachicha committed:
add formatted code

nhachicha committed:
Merge branch 'ic/export-schemas' of https://github.com/realm/realm-studio into ic/export-schemas

nhachicha committed:
Update README, merged origin

nhachicha committed:
add missing file

IvanCoronado committed:
refactor (MainMenu): Use exportSchema function for Java

Commit:
Merge branch 'master' of github.com:realm/realm-studio into ic/export-schemas

kraenhansen reviewed (dismissed):
As soon as the tests pass, I believe this should get merged in and released.

nhachicha committed:
fixing tests

nhachicha committed:
automatic formatted code

nhachicha committed:
Merge branch 'ic/export-schemas' of https://github.com/realm/realm-studio into ic/export-schemas

Commit:
Merge branch 'ic/export-schemas' of github.com:realm/realm-studio into ic/export-schemas

Commit:
Reverted the eventListenerCallbacks

I figured that these should have been in the options object - but it
doesn't feel practical

Commit:
Taking a more functional approach on the menu updating

I believe mutating the menu items are generally disincuraged

Commit:
Using enum values that are suitable for displaying

Commit:
Got a TS error on this line

Commit:
Registering the listener on mount + removing it on unmount

Commit:
Adding a message + a defaultPath to the export schema save dialog

kraenhansen reviewed (approved):
I made a few changes - anticipating that we soon will be adding more menu items on the different window types.
